client/gutenberg/extensions/videopress/editor.js

```diff
@@ -8,12 +8,56 @@ import { addFilter } from '@wordpress/hooks';
 /**
  * Internal dependencies
  */
-import isJetpackExtensionAvailable from 'gutenberg/extensions/presets/jetpack/utils/is-jetpack-extension-available';
 import withVideoPressEdit from './edit';
+import withVideoPressSave from './save';
+import { isEnabled } from 'config';
 
-const addVideoPressSupport = ( settings, name ) =>
-	'core/video' === name ? { ...settings, edit: withVideoPressEdit( settings.edit ) } : settings;
+const addVideoPressSupport = ( settings, name ) => {
+	if ( 'core/video' !== name ) {
+		return settings;
+	}
 
-if ( isJetpackExtensionAvailable( 'videopress' ) ) {
+	return {
+		...settings,
+		attributes: {
+			...settings.attributes,
+			guid: {
+				type: 'string',
+			},
+			autoplay: {
+				type: 'boolean',
+			},
+			controls: {
+				type: 'boolean',
+				default: true,
+			},
+			loop: {
+				type: 'boolean',
+			},
+			muted: {
+				type: 'boolean',
+			},
+			poster: {
+				type: 'string',
+			},
+			preload: {
+				type: 'string',
+				default: 'metadata',
+			},
+		},
+		edit: withVideoPressEdit( settings.edit ),
+		save: withVideoPressSave( settings.save ),
+		deprecated: [
```

Comment:
Not sure why, but every time I open an existing post containing a video block I see this error when the block is migrated to the customized version. @gziolo @aduth @youknowriad do you have any idea why the block validation fails? The actual content seems correct to me, so I don't know why the validation expects a different content.

<img width="1006" alt="screen shot 2019-02-07 at 13 14 52" src="https://user-images.githubusercontent.com/1233880/52411048-af12e900-2ada-11e9-841a-c5bdecf8866c.png">

Reply:
It might be worth referencing documentation on how block invalidation works:

https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-edit-save/#validation

The message indicates that the editor is concerned about a possibly destructive action, because what it generates is different than what was originally saved.

In your case, I'm assuming it's because you're adding new DOM attributes (autoplay, loop) in the `save` implementation? I'm not sure what basis there was for extending the core blocks in this fashion, but it's going to be highly prone to these sorts of validation errors (and problematic if the content is ever exported to a site which isn't "VideoPress-enhanced").

Reply:
> In your case, I'm assuming it's because you're adding new DOM attributes (autoplay, loop) in the `save` implementation?

I'm not adding new attribute, so that's why I don't understand the error. 

The current `save` implementation of the video block include these DOM attributes ([#](https://github.com/WordPress/gutenberg/blob/c5f9bd88125282a0c35f887cc8d835f065893112/packages/block-library/src/video/index.js#L116-L122)). The other difference is the `wp-block-video` class, but again that is added by the `save` implementation (using the [`blocks.getSaveContent.extraProps` filter](https://github.com/WordPress/gutenberg/blob/c5f9bd88125282a0c35f887cc8d835f065893112/packages/editor/src/hooks/generated-class-name.js#L42)).

Maybe the expected content is generated from a different `save` implementation that is not considering DOM attributes and hooks?

Reply:
Are you still seeing this @mmtr ? I couldn't reproduce when loading multiple times from Gutenlypso.

Reply:
Oh hmm the new props aren't being added in the html comment, and they don't appear in block options 🤔

Reply:
See my comment on https://github.com/Automattic/wp-calypso/pull/30546#issuecomment-462912103